cumulus/pallets/xcmp-queue/src/lib.rs

```diff
@@ -791,22 +788,41 @@ impl<T: Config> XcmpMessageHandler for Pallet<T> {
 							},
 						}
 					},
-				XcmpMessageFormat::ConcatenatedVersionedXcm =>
+				XcmpMessageFormat::ConcatenatedVersionedXcm => {
+					let mut new_page = true;
 					while !data.is_empty() {
 						let Ok(xcm) = Self::take_first_concatenated_xcm(&mut data, &mut meter)
```

Comment:
Does it make sense to use the same pattern also here here for `if meter.try_consume(T::WeightInfo::take_first_concatenated_xcm())`?
```suggestion
						let Ok(xcm) = Self::take_first_concatenated_xcm(&mut data)
```

Reply:
Here I would prefer to keep it as before since we don't care about new pages and  it's easy to do the weight metering inside the function. For `enqueue_xcmp_message` it seemed strange to add a parameter `new_page` just for weight metering. 

But if you have strong preferences I can change it.